### PR TITLE
NFC: green tea test for EEPROM driver

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
### Description

Green tea test for EEPROM drivers, testing all common operations but not doing any stress testing or testing of large data (I'm waiting for Don's code for that to avoid repeating the same logic).

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] New target
    [x] Feature
    [ ] Breaking change

### CODE REVIEW

File: TESTS/nfc/eeprom/main.cpp [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+
+/* defines the "script" the test follows, each step is checked and all
+ * have to execute to completion for a successful run */
+static const TestCommand_t SEQUENCES[SEQUENCE_MAX][MAX_STEP] = {
+    /* SESSION_TEST */
+    { START_SESSION, END_SESSION, TERMINATE },
+
+    /* WRITE_READ_TEST */
+    { START_SESSION, WRITE_SIZE_4, WRITE_4_BYTES, READ_4_BYTES, WRITE_2_BYTES,
+      READ_4_BYTES_MIXED, END_SESSION, TERMINATE },
```
Comment (donatieng) [inline TESTS/nfc/eeprom/main.cpp]: Here and in the erase test, you write the size but don't read it back?

**Review by donatieng — CHANGES_REQUESTED** [review]

File: TESTS/nfc/eeprom/main.cpp (hunk 1 above)
Comment (paul-szczepanek-arm) [reply TESTS/nfc/eeprom/main.cpp]: yes, size changes are tested elsewhere, this size write is part of the writing of bytes which is tested by reading them back, no harm checking the size though

**Comment by paul-szczepanek-arm** [comment]
@donatieng does this resolve your request?

**Review by donatieng — APPROVED** [review]
Thanks @paul-szczepanek-arm !

**Pull request merged**